lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+     */
+    callbackParams (input: string | IncomingMessage | Http2ServerRequest): CallbackParamsType
+
+    /**
+     * Performs the callback for Authorization Server's authorization response.
+     * @param redirectUri redirect_uri used for the authorization request
+     * @param parameters returned authorization response, see client.callbackParams if you need help getting them.
+     * @param checks
+     * @param extras
+     */
+    callback(redirectUri: string, parameters: CallbackParamsType, checks?: IExtendedCallbackChecks, extras?: ICallbackExtras) : Promise<TokenSet>
+
+    /**
+     * Pure OAuth 2.0 version of callback().
+     * @param redirectUri
+     * @param parameters
+     * @param checks
+     * @param extras
+     */
+    oauthCallback(redirectUri: string, parameters: CallbackParamsType, checks?: ICallbackChecks, extras?: ICallbackExtras) : Promise<TokenSet>
+
+    /**
+     * Performs refresh_token grant type exchange.
+     * @param refreshToken Refresh Token value. When TokenSet instance is provided its refresh_token property
+     * will be used automatically.
+     * @param extras
+     */
+    refresh(refreshToken: TokenType, extras?: IRefreshExtras) : Promise<TokenSet>
+
+    /**
+     * Fetches the OIDC userinfo response with the provided Access Token. Also handles signed and/or
+     * encrypted userinfo responses. When TokenSet is provided as an argument the userinfo sub property
+     * will also be checked to match the on in the TokenSet's ID Token.
+     *
+     * @param accessToken Access Token value. When TokenSet instance is provided its access_token property
+     * will be used automatically.
+     */
+    userinfo(accessToken: TokenType) : Promise<object>
+
+    /**
+     * Performs an arbitrary grant_type exchange at the token_endpoint.
+     */
+    grant(body: IGrantBody, extras?: IGrantExtras) : Promise<TokenSet>
+
+    /**
+     * Introspects a token at the Authorization Server's introspection_endpoint.
+     */
+    introspect(token: string, tokenTypeHint?: string, extras?: IIntrospectExtras) : Promise<object>
+
+    /**
+     * Revokes a token at the Authorization Server's revocation_endpoint.
+     */
+    revoke(token, tokenTypeHint?: string, extras?: IRevokeExtras) : Promise<void>
+
+    /**
+     * Creates a signed and optionally encrypted Request Object to send to the AS. Uses the client's
+     * request_object_signing_alg, request_object_encryption_alg, request_object_encryption_enc metadata for
+     * determining the algorithms to use.
+     */
+    requestObject(payload: IRequestObjectPayload): Promise<string>
+
+    /**
+     * Starts a Device Authorization Request at the issuer's device_authorization_endpoint and returns a handle
+     * for subsequent Device Access Token Request polling.
+     */
+    deviceAuthorization(parameters?: IDeviceAuthParameters, extras?: IDeviceAuthExtras): Promise<IDeviceFlowHandle>
+
+    [key: string]: any
```

Comment:
Are you sure we want this extension here? We shouldn't be assuming there are additional properties on a client